src/blockchain/download-augur-logs.ts

```diff
@@ -31,3 +34,13 @@ export function downloadAugurLogs(db: Knex, augur: Augur, fromBlock: number, toB
     ), callback);
   });
 }
+
+function downloadBlocksFromLogs(db: Knex, augur: Augur, allAugurLogs?: AugurLogs): void {
+  const blockNumbers = _.uniq(
```

Comment:
Unless Lodash has some fancy optimization to avoid this, this `blockNumbers` calculation will generate an array with an entry for each log, from which a smaller array w/ just the unique numbers is extracted.  Possibly more efficient to define `const blockNumbers = {}` and add a line inside the nested `eachSeries`:
```javascript
if (!blockNumbers[log.blockNumber]) blockNumbers[log.blockNumber] = true;
```
and then use `Object.keys(blockNumbers)` to loop over the numbers to pass to `processBlockByNumber`.

(Possibly a premature optimization, just jumped out at me a bit when reading.)

Reply:
I was trying to avoid having unlinked data in the database, where we're inserting the log events before the blocks (and hence would need the callback) but maybe it isn't that important.

The way i've done it does feel a bit too pythonic to me, i'll change.